docs/rfcs/XeTile.md

```diff
@@ -185,12 +232,275 @@ For example, for the tile size [128, 128] and sg_data [32, 128], along the secon
+
+This is a use case for the order attribute of XeTile.tile. In this use case, the matrix B has a transposed memory layout to start with, for example BT [N,K] instead of B[K, N].  But the user likes  to use the original  program to index the matrix as if it is B[K, N], the order attribute is introduced to support this programming. User can flip the 2d block offset and size, and swap the stride from [K, 1] to [1, K].
+```mlir
+   BF16 A[M][K], BT[N, K], C[M][N];    // C = MM(A, BT)
+   For i = 0, M-1, M_tile  Do
+    For j = 0, N-1, N_tile Do
+        For k = 0, K-1, K_tile  Do
+            %a = init_tile &A, [i, k], [M, K], [K, 1] : tile<64x32xbf16>;                	// M_tile=64, K_tile=32
+            %b = init_tile &BT, [k, j], [K, N], [1, K] : tile<32x64xbf16, order = [0, 1]>;  	// K_tile =32, N_tile=64
+            %c = init_tile &C, [i, j], [M, N], [N, 1] : tile<64x64xbf16>;               	// M_tile=64, N_tile=64
+            %va = load_tile %a : vector<64x32xbf16>;
+             %vb = load_tile %b : vector<32x64x bf16>;
+             %vc = tile_mma %va, %vb : vector<64x32xbf16>, vector<32x64x bf16> into vector<64x64xbf16>;
+```
+
+Alternatively, the user may just writes the program according to the given memory layout but apply a tile_transpose after the code being loaded. This is also an valid code.
+```
+BF16 A[M][K], BT[N, K], C[M][N];    // C = MM(A, BT)
+For i = 0, M-1, M_tile  Do
+    For j = 0, N-1, N_tile Do
+        For k = 0, K-1, K_tile  Do
+            %a = init_tile &A, [i, k], [M, K], [K, 1] : tile<64x32xbf16>;                // M_tile=64, K_tile=32
+            %bt = init_tile &BT, [j, k], [N, K], [K, 1] : tile<64x32xbf16>;  	 	// N_tile=64, K_tile=32
+            %c = init_tile &C, [i, j], [M, N], [N, 1] : tile<64x64xbf16>;               // M_tile=64, N_tile=64
+             %va = load_tile %a : vector<64x32xbf16>;
+             %vbt = load_tile %bt : vector<64x 32x bf16>;
+             %vb = tile_transpose %vbt: vector<64x32xbf16> into  vector<32x64x bf16>;
+             %vc = tile_mma %va, %vbt : vector<64x32xbf16>, vector<32x64x bf16> into vector<64x64xbf16>;
+```
+
+All these three use cases can be programed by using memref and vector dialect. User may run into the same issue that matrix B is given as BT, so it is presented as the memory as a transposed matrix. User also have the same two choices to write the program, either use the plain layout memref reflecting the physical memory layout (3rd use case), or try to use the stride or affine_map attribute to represent it as “col-major” layout  (2nd use case).   Memref has a stride and affine_map attribute, both can used to describe the memory layout. So a memref a[i, j] could be refer to the position to i*stride_i + j*stride_j (using stride), j*stride_i + i (using affien_map to swap index).  This effectively creates the same effect that order[0, 1] attribute try to provide to user. User now can swap the i, j in the program.
```

Comment:
I guess it's better "...it is presented IN the memory as a transposed matrix."

Reply:
fxied